### ISSUE TITLE
"No attribute dispersion" errors when writing spectrum with tabular WCS

### ISSUE THREAD

**Issue by parejkoj** [astropy/specutils#528]
I've tried to construct a relatively simple 1d spectrum with a per-pixel wavelength solution using specutils (0.6.0) but I can't output it as a fits file. I used two approaches, both cobbled together from what I could glean from various examples. First I tried sending the wavelength solution as the `spectral_axis`:

```python
import astropy.units as u
import specutils

n_pixels=1000
wavelength = np.linspace(300, 1100, n_pixels)*u.nm
spectrum = np.random.random(n_pixels) * u.count

spec=specutils.Spectrum1D(flux=spectrum, spectral_axis=wavelength)
```

and also creating a gwcs `SpectralFrame` and a `Tabular1d` model and passing that in as the `wcs` kwarg:

```python
import astropy.units as u
import specutils
import gwcs
from astropy.modeling.tabular import Tabular1D

n_pixels=1000
wavelength = np.linspace(300, 1100, n_pixels)*u.nm
spectrum = np.random.random(n_pixels) * u.count

forward = Tabular1D(np.arange(n_pixels), lookup_table=wavelength, name='dispersion')
forward.inverse = Tabular1D(wavelength, lookup_table=wavelength)
pixel_frame = gwcs.CoordinateFrame(naxes=1, axes_type=('SPECTRAL',), axes_order=(0,))
spectral_frame = gwcs.SpectralFrame(axes_order=(0,), unit=wavelength.unit)
wcs = gwcs.wcs.WCS(forward_transform=forward, input_frame=pixel_frame, output_frame=spectral_frame)

spect = specutils.Spectrum1D(flux=spectrum, wcs=wcs)
```

In either case, the writer cannot identify the format when I try to write the output to a FITS file, and I get a different error if I try to specify the format: both `wcs1d_fits_writer` and `tabular_fits_writer` assume there is a `dispersion` attribute on the spectrum, but I don't see where to specify that when I construct the Spectrum1D.

```
In [2]: spec.write('testing.fits')

~/lsst/lsstsw/miniconda/envs/lsst-scipipe/lib/python3.7/site-packages/astropy/io/registry.py in _get_valid_format(mode, cls, path, fileobj, args, kwargs)
    593         raise IORegistryError("Format could not be identified.\n"
    594                               "The available formats are:\n"
--> 595                               "{0}".format(format_table_str))
    596     elif len(valid_formats) > 1:
    597         raise IORegistryError(

IORegistryError: Format could not be identified.
The available formats are:
   Format    Read Write Auto-identify
------------ ---- ----- -------------
tabular-fits  Yes   Yes           Yes
    wcs-fits   No   Yes            No

In [3]: spec.write('testing.fits', format='wcs-fits')

AttributeError: 'Spectrum1D' object has no attribute 'dispersion'

In [4]: spec.write('testing.fits', format='tabular-fits')                       

~/lsst/lsstsw/miniconda/envs/lsst-scipipe/lib/python3.7/site-packages/specutils/io/default_loaders/tabular_fits.py in tabular_fits_writer(spectrum, file_name, **kwargs)
     75 def tabular_fits_writer(spectrum, file_name, **kwargs):
     76     flux = spectrum.flux.value
---> 77     disp = spectrum.dispersion.value
     78     meta = spectrum.meta

AttributeError: 'Spectrum1D' object has no attribute 'dispersion'
```

**Comment by nmearl** [comment]
It seems like the writers have not been updated with changes made to the `Spectrum1D` class. As a workaround for now, you can [create your own loader](https://specutils.readthedocs.io/en/stable/custom_loading.html#creating-a-custom-writer). Of course, now that I look at the docs, it too is using `dispersion`, so the next point release of specutils will have these fixes. But you'd copy the same formalism, but use `spectral_axis` instead of `dispersion`.

**Comment by parejkoj** [comment]
Separately, it looks like the `wcs1d_fits_writer` and `tabular_fits_writer` are actually identical. We should be able to get rid of one of them: I'd suggest wcs1d, since it doesn't look like it actually writes a wcs.

**Comment by nmearl** [comment]
This issue was solved with PR #528. Please comment if this is not the case.
